src/core_editor/editor.rs

```diff
@@ -626,20 +626,35 @@ impl Editor {
     /// If a selection is active returns the selected range, otherwise None.
     /// The range is guaranteed to be ascending.
     pub fn get_selection(&self) -> Option<(usize, usize)> {
+        // Default to Vi-style behavior for backward compatibility
+        self.get_selection_with_mode(&PromptEditMode::Vi(PromptViMode::Normal))
```

Comment:
Is there an explicit need to be backwards compatible here?

Maybe the correct thing would be to split it into true selection (which covers the characters in a consistent way with a range using indices used by the edit ops), and visual selection which is mode aware.

It seems the bug prone part was the visual selection, if there are no other users in reedline itself we should consider what the right behavior would be for a public user of `get_selection` (and if that is unlikely we can make things private.